### DISCUSSION TITLE
[Kavita] How are covers selected and/or generated?

### DISCUSSION THREAD

**Discussion by Dragonatorul** [Kareadita/Kavita#2717] [Ideas]
### Idea Description

I couldn't tell from the wiki documentation how a cover is selected for a given series and/or chapter. 

- What is the ideal way to structure a file so that Kavita uses a specific image as the cover?
- Is there a way to define a specific image outside of a cbz as the series cover?

To be specific, I want to be able to have a folder with a set of CBZ files and define the cover image for the series, ideally without having to include it in one of the CBZ files. 

If this is not supported here are some ideas:

- Extract the info from the ComicInfo `ComicPageInfo` metadata
- Use a specific filename in the CBZ (like `series_cover.[ext]`
- Use a specific file at the series root folder level. 

### Duration of Using Kavita

2 years

### Idea Category

Feature Enhancement

### Agreement

- [X] I agree that this is solely for submitting ideas, and I will search for existing ideas before posting.

**Comment by majora2007** [comment]
Covers are selected by selecting the first file in the archive that has the word cover else defaulting to the first image sorted naturally.

Kavita doesn't use the comicinfo cover image designation and has no plans to, due to many people not using it. 

Sidecar covers is not something I'm a fan of. I do agree it's the easiest for users to not have to modify their files, but it's not something I personally want to maintain support for. 

Kavita does have an API which you can use to save the cover in the system so that you can get custom covers without modifying the underlying archive.

**Reply by LeoShivas** [reply]
Do you have an example of how to acheive this by calling the API ?

**Reply by scare376** [reply]
https://www.kavitareader.com/docs/api/
